ingredient/ingredient-apim-api/src/functions.ts

```diff
@@ -21,4 +21,20 @@ export class ApimApiUtils extends BaseUtility {
 
         return backend;
     }
+
+    public get_hostheader(namespace: string, k8sHostname: string, serviceName: string | null = null): string {
+        var hostHeader = `${serviceName || this.context.Config.shortName}-${namespace}.${k8sHostname}`;
```

Comment:
if serviceName is null, should we fallback to using package short name?

Reply:
ya since thats what istio uses for the host header. 99% of time `serviceName` should be nullb but there in case anyone has a special case.

Reply:
should that fallback be done in get_hostheader?

Reply:
as opposed to where? Im not against not having it or moving it since it feels like a very "HCHB" in an open source project. But then again we also have a property service ingredient and other pretty specific things :)

Reply:
I would add it here, or add a function that can do create the serviceName for you, passing in an optional name. Can then pass that into get_hostheader.

Reply:
it is here already `${serviceName || this.context.Config.shortName}` maybe thinking perhaps you didn't see it from the context im getting from the comments? :)

Reply:
yeah I'm dumb